bluetooth/src/commonMain/kotlin/scanner/Scanner.kt

```diff
@@ -224,6 +232,30 @@ abstract class BaseScanner constructor(
         }
     }
 
+    override suspend fun retrievePairedDevices(withServices: Set<UUID>) {
```

Comment:
I would consider splitting it into a few methods with readable names or at least adding some more comments